src/raftstore/store/peer.rs

```diff
@@ -1435,6 +1435,10 @@ impl Peer {
         None
     }
 
+    pub fn approximate_size(&self) -> Result<u64> {
```

Comment:
Don't add the calculation into `raftstore` thread, it might block raftstore thread.